desktop/ui/src/main/java/org/datacleaner/widgets/tree/SchemaTree.java

```diff
@@ -647,7 +647,7 @@ public Datastore getDatastore() {
     }
 
     @Override
-    public void componentDescriptorsUpdated() {
+    public synchronized void componentDescriptorsUpdated() {
```

Comment:
Actually the issue here is not really one of concurrency, but one of worker thread responsibility. You need to ensure that the update of the tree happens in the Swing UI thread instead of whatever thread is now invoking this.

To do that, use the following:
```
WidgetUtils.invokeSwingAction(new Runnable() {
  // any UI modification code goes into the runnable here
});
```